src/utils/misc/misc.ts

```diff
@@ -188,3 +200,54 @@ export const formatBoolean = (value: boolean) => (value ? "true" : "false");
  */
 export const formatNilBoolean = (value: Nil<boolean>) =>
   isNil(value) ? null : formatBoolean(value);
+
+/**
+ * Joins path segments, handling leading/trailing slashes between segments.
+ * It accepts strings and numbers; ignores null/undefined/empty segments.
+ *
+ * @example
+ * joinPaths("/api/v1/", "/users", "123"); // "/api/v1/users/123"
+ *
+ * @example
+ * joinPaths("api", "users"); // "api/users"
+ *
+ * @example
+ * joinPaths("https://example.com/", "/users/", "123"); // "https://example.com/users/123"
+ *
+ * @example
+ * joinPaths("/a", "b?x=1#top"); // "/a/b?x=1#top"
+ */
+export const joinPaths = (
+  ...segments: Array<string | number | null | undefined>
+): string => {
```

Comment:
Can we remove return type? It's easily inferable. 

```suggestion
) => {
```